Tests/Unit/HttplugFactoryTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Http\HttplugBundle\Tests\Unit\ClientFactory;
+
+use Http\HttplugBundle\Collector\MessageJournal;
+use Http\HttplugBundle\HttplugFactory;
+use Puli\Discovery\Api\Discovery;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttplugFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testEvaluateCondition()
+    {
+        $method = new \ReflectionMethod(HttplugFactory::class, 'evaluateCondition');
```

Comment:
The correct way would be mocking the Puli discovery, returning a mocked Class Binding and using it for testing here.